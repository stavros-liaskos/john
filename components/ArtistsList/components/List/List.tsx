import LastFm from '../../../Icons/lastfm';
import Spotify from '../../../Icons/spotify';
import Button from '../../../Button/Button';
import React from 'react';
import { ArtistsListProp } from '../../ArtistsList';

const ICON_SIZE = 30;

const List = ({ i18n, artistsList, onButtonClick, artistLoading }: ArtistsListProp) => {
  if (!i18n || !i18n.btnTxt || !i18n.noArtists || !artistsList?.length || !onButtonClick) {
    return null;
  }

  return (
    <>
      {artistsList.map((artist, index: number) => (
        <div
          className="flex justify-between md:justify-center items-center dark:even:bg-gh-darkly even:bg-gray-100"
          key={artist.id}
        >
          <p className="grow text-clip rr-text">{artist.name}</p>
          <div className="flex basis-2 mx-4 md:mx-8">
            {artist.lastfmUri && (
              <a className="inline" href={artist.lastfmUri}>
                <LastFm width={ICON_SIZE} height={ICON_SIZE} />
              </a>
            )}
            {artist.spotifyUri && (
              <a className="inline" href={artist.spotifyUri}>
                <Spotify width={ICON_SIZE} height={ICON_SIZE} />
              </a>
            )}
          </div>
          <Button
            i18n={i18n.btnTxt}
            handleClick={onButtonClick}
            handleClickArg={artist.id}
            className={`btn-small lg:ml-8 my-2 ${index % 2 ? '!border-gh-dark' : ''}`}
            disabled={!!artistLoading && artist.id === artistLoading}
            loading={!!artistLoading && artist.id === artistLoading}
          />
        </div>
      ))}
    </>
  );
};

export default List;
