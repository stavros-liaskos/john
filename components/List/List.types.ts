export interface ListProps {
  i18n: { unfollow: string };
}
