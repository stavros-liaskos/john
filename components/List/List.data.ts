import { ListEl } from './List.types';

export const listData: ListEl[] = [
  {
    followedByUser: false,
    lastfmUri: 'https://www.last.fm/music/Liedfett',
    name: 'Liedfett',
    spotifyUri: '',
  },
  {
    followedByUser: false,
    lastfmUri: 'https://www.last.fm/music/Leaders+of+the+New+School',
    name: 'Leaders of the New School',
    spotifyUri: '',
  },
  {
    followedByUser: false,
    lastfmUri: 'https://www.last.fm/music/lo+&+leduc',
    name: 'lo & leduc',
    spotifyUri: '',
  },
];

export const listI18n = {
  unfollow: 'unfollow',
};
