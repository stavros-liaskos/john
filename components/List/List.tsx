import Image from 'next/image';
import { ListProps, ListEl } from './List.types';
import React from 'react';

const List: React.FunctionComponent<ListProps> = ({ list }) => {
  if (!list) {
    return null;
  }

  return (
    <div className="border border-indigo-100">
      {list.map((el: ListEl, index: number) => (
        <div className="max-w-md mx-auto bg-white rounded-xl shadow-md overflow-hidden md:max-w-2xl" key={index}>
          <div className="md:flex">
            <div className="md:shrink-0">
              <Image
                className="h-48 w-full object-cover md:h-full md:w-48"
                src="https://www.placecage.com/120/120"
                alt={`Logo of ${el.name}`}
                width={120}
                height={170}
              />
            </div>
            <div className="p-8">
              <div className="uppercase tracking-wide text-sm text-indigo-500 font-semibold">Case study</div>
              <a href="#" className="block mt-1 text-lg leading-tight font-medium text-black hover:underline">
                {el.name}
              </a>
              <p className="mt-2 text-gray-500">
                Getting a new business off the ground is a lot of hard work. Here are five ideas you can use to find
                your first customers.
              </p>
            </div>
          </div>
        </div>
      ))}
    </div>
  );
};

export default List;
