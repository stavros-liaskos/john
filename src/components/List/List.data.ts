import { ListEl } from './List.types';

export const listData: ListEl[] = [
  {
    name: 'Led Zeppeling',
    spotifyUrl: 'www.google.com',
  },
  {
    name: 'YOYOO',
    lastFmUrl: 'www.google.com',
  },
  {
    name: 'Art Beckley',
    lastFmUrl: 'www.google.com',
    spotifyUrl: 'www.google.com',
  },
];

export const listI18n = {
  unfollow: 'unfollow',
};
