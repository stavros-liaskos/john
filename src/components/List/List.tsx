import React from 'react';
import { ListProps, ListEl } from './List.types';
import Spotify from '../Icons/spotify';
import LastFm from '../Icons/lastfm';

const List: React.FunctionComponent<ListProps> = ({ list, i18n }) => {
  if (!list || !i18n || !i18n.unfollow) {
    return null;
  }

  return (
    <div className="border border-indigo-100">
      {list.map((el: ListEl, index: number) => (
        <div className="flex justify-between items-center px-8 even:bg-gray-100" key={index}>
          <p>{el.name}</p>
          <div className="flex">
            {el.lastFmUrl && (
              <a className="inline" href={el.lastFmUrl}>
                <LastFm />
              </a>
            )}
            {el.spotifyUrl && (
              <a className="inline" href={el.spotifyUrl}>
                <Spotify />
              </a>
            )}
          </div>
          <button className="py-2 px-3 bg-indigo-800 text-white text-sm font-semibold rounded-md shadow focus:outline-none">
            {i18n.unfollow}
          </button>
        </div>
      ))}
    </div>
  );
};

export default List;
