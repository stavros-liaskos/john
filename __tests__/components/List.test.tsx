import React from 'react';
import { getAllByRole, render } from '@testing-library/react';
import List from '../../components/List/List';
import { listData } from '../../components/List/List.data';

describe('List', () => {
  it('renders without data without crashing', () => {
    // @ts-ignore
    render(<List />);
  });

  it('lists 2 artists', () => {
    const { container } = render(<List list={listData} />);
    const artistLogos = getAllByRole(container, 'img');

    expect(artistLogos.length).toEqual(2);
    expect(container).toMatchSnapshot();
  });
});
