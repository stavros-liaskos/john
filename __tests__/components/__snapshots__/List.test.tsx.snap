// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List lists 2 artists 1`] = `
<div>
  <div
    class="border border-indigo-100"
  >
    <div
      class="max-w-md mx-auto bg-white rounded-xl shadow-md overflow-hidden md:max-w-2xl"
    >
      <div
        class="md:flex"
      >
        <div
          class="md:shrink-0"
        >
          <span
            style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
          >
            <span
              style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
            >
              <img
                alt=""
                aria-hidden="true"
                src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTIwIiBoZWlnaHQ9IjE3MCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
                style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
              />
            </span>
            <img
              alt="Logo of Led Zeppeling"
              class="h-48 w-full object-cover md:h-full md:w-48"
              data-nimg="intrinsic"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
            <noscript />
          </span>
        </div>
        <div
          class="p-8"
        >
          <div
            class="uppercase tracking-wide text-sm text-indigo-500 font-semibold"
          >
            Case study
          </div>
          <a
            class="block mt-1 text-lg leading-tight font-medium text-black hover:underline"
            href="#"
          >
            Led Zeppeling
          </a>
          <p
            class="mt-2 text-gray-500"
          >
            Getting a new business off the ground is a lot of hard work. Here are five ideas you can use to find your first customers.
          </p>
        </div>
      </div>
    </div>
    <div
      class="max-w-md mx-auto bg-white rounded-xl shadow-md overflow-hidden md:max-w-2xl"
    >
      <div
        class="md:flex"
      >
        <div
          class="md:shrink-0"
        >
          <span
            style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
          >
            <span
              style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
            >
              <img
                alt=""
                aria-hidden="true"
                src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTIwIiBoZWlnaHQ9IjE3MCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
                style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
              />
            </span>
            <img
              alt="Logo of Led Zeppeling"
              class="h-48 w-full object-cover md:h-full md:w-48"
              data-nimg="intrinsic"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
            <noscript />
          </span>
        </div>
        <div
          class="p-8"
        >
          <div
            class="uppercase tracking-wide text-sm text-indigo-500 font-semibold"
          >
            Case study
          </div>
          <a
            class="block mt-1 text-lg leading-tight font-medium text-black hover:underline"
            href="#"
          >
            Led Zeppeling
          </a>
          <p
            class="mt-2 text-gray-500"
          >
            Getting a new business off the ground is a lot of hard work. Here are five ideas you can use to find your first customers.
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
